### PR TITLE
5.3 (in progress)

### PR DESCRIPTION
**Pull request by Nathat23**

### CODE REVIEW

**Comment by Aurelien30000** [comment]
Listening to ChunkLoadEvent is sadly a bad idea because this event is fired a lot on Spigot and even more on Paper according to Aikar. I don't think it's a reliable solution for performance issues :/

**Comment by Nathat23** [comment]
I mean I guess it could just be removed if needed? We could only use EntityRemoveFromWorldEvent (with fallback for spigot users)? Or check if mobs are still loaded in the stacking task?

**Comment by Aurelien30000** [comment]
> I mean I guess it could just be removed if needed? We could only use EntityRemoveFromWorldEvent (with fallback for spigot users)? Or check if mobs are still loaded in the stacking task?

Certainly

File: src/main/java/uk/antiperson/stackmob/StackMob.java [hunk 1]
```diff
@@ -141,7 +147,7 @@ private void registerEvents() throws InvocationTargetException, NoSuchMethodExce
         registerEvent(TargetListener.class);
         registerEvent(PlayerListener.class);
         registerEvent(BeeListener.class);
-        registerEvent(LeashListener.class);
```
Comment (Aurelien30000) [inline src/main/java/uk/antiperson/stackmob/StackMob.java:144]: Why did you remove this?

Comment (Nathat23) [reply src/main/java/uk/antiperson/stackmob/StackMob.java:144]: It was accidental, I added it back in a later commit 6ca24828707b304c167c78fe4fb8d17739a76f61

File: pom.xml [hunk 2]
```diff
@@ -100,7 +100,7 @@
         <dependency>
             <groupId>org.spigotmc</groupId>
             <artifactId>spigot</artifactId>
-            <version>1.16.1-R0.1-SNAPSHOT</version>
+            <version>1.16.2-R0.1-SNAPSHOT</version>
             <scope>provided</scope>
```
Comment (Aurelien30000) [inline pom.xml:104]: You should also update paper, worldguard and worldedit :)

```
        <dependency>
            <groupId>com.destroystokyo.paper</groupId>
            <artifactId>paper-api</artifactId>
            <version>1.16.2-R0.1-SNAPSHOT</version>
            <scope>provided</scope>
        </dependency>
        <dependency>
            <groupId>org.spigotmc</groupId>
            <artifactId>spigot</artifactId>
            <version>1.16.2-R0.1-SNAPSHOT</version>
            <scope>provided</scope>
        </dependency>
        <dependency>
            <groupId>com.sk89q.worldguard</groupId>
            <artifactId>worldguard-bukkit</artifactId>
            <version>7.0.4-SNAPSHOT</version>
            <scope>provided</scope>
        </dependency>
        <dependency>
            <groupId>com.sk89q.worldedit</groupId>
            <artifactId>worldedit-bukkit</artifactId>
            <version>7.2.0-SNAPSHOT</version>
            <scope>provided</scope>
        </dependency>
```

File: src/main/java/uk/antiperson/stackmob/tasks/TagTask.java [hunk 3]
```diff
@@ -34,13 +34,13 @@ public void run() {
                 if (!sm.getEntityManager().isStackedEntity((LivingEntity) entity)) {
                     continue;
                 }
+                StackEntity stackEntity = sm.getEntityManager().getStackEntity((LivingEntity) entity);
```
Comment (Aurelien30000) [inline src/main/java/uk/antiperson/stackmob/tasks/TagTask.java:37]: Useless currently, you don't need the object yet, should be reverted unless you'll need it in future ^^

File: src/main/java/uk/antiperson/stackmob/entity/EntityManager.java [hunk 4]
```diff
@@ -1,25 +1,103 @@
...
+
+    public void registerAllEntities() {
+        for (World world : Bukkit.getWorlds()) {
+            for (Chunk chunk : world.getLoadedChunks()) {
+                registerStackedEntities(chunk);
+            }
+        }
+    }
+
+    public void unregisterAllEntities() {
```
Comment (Aurelien30000) [inline src/main/java/uk/antiperson/stackmob/entity/EntityManager.java:49]: If we got rid of the chunk listener, is this still required?

Comment (Nathat23) [reply src/main/java/uk/antiperson/stackmob/entity/EntityManager.java:49]: Not really.
